core/trino-main/src/main/java/io/trino/operator/Driver.java

```diff
@@ -524,6 +534,42 @@ private ListenableFuture<Void> processInternal(OperationTimer operationTimer)
         return NOT_BLOCKED;
     }
 
+    private void validateOperatorOuputPage(Page page, Operator operator, List<Type> types)
+    {
+        if (page.getChannelCount() != types.size()) {
+            throw new TrinoException(
+                    GENERIC_INTERNAL_ERROR,
+                    "Invalid number of channels; got %s expected %s; %s".formatted(page.getChannelCount(), types.size(), operatorDebugInfo(operator)));
```

Comment:
Worth printing all expected types and all actual block types just as if there was a type mistmatch